### PR TITLE
add expr package

### PR DESCRIPTION
**Pull request by ozym**
This is a helper package that makes adding numerical configuration where it makes more sense to use an equation rather than the final answer.

### CODE REVIEW

File: internal/expr/expr.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	return res, nil
+}
+
+// ToFloat64 evaluates the given string as a float64, or it returns an error if the result is invalid or the expression is Unknown.
+func ToFloat64(s string) (float64, error) {
+	val, err := eval(s)
+	if err != nil {
+		return 0, err
+	}
+	res, _ := constant.Float64Val(val)
```
Comment (bpeng) [inline internal/expr/expr.go:42]: Does it make sense to validate the type before getting the value, to avoid possible panic?

**Review by bpeng — COMMENTED** [review]
Looks good, little suggestion which may not be necessary

File: internal/expr/expr.go (hunk 1 above)
Comment (ozym) [reply internal/expr/expr.go:42]: I think, to some extent this is already handled by the underlying code.

e.g.
https://cs.opensource.google/go/go/+/refs/tags/go1.17.7:src/go/constant/value.go;l=548

Comment (ozym) [reply internal/expr/expr.go:42]: but yes, there's a panic in the.

**Comment by ozym** [comment]
@bpeng I have added some guardrails around the values after evaluating, this is a good enhancement as I suspect this may have caused issues when dealing with floating point values that are given as integers.

I've also added a complex number value, as this is something that crops up a lot in response configuration.

**Review by bpeng — APPROVED** [review]
OK

**Pull request merged**